### PR TITLE
migrate sharedb to new plugin system

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
migrate sharedb to new plugin system

### Motivation
migrate plugins to new plugin system

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/datadog-instrumentations/src/sharedb.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  shimmer.wrap(Agent.prototype, '_handleMessage', origHandleMessageFn => function (request, callback) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+    const action = request.a
+
+    const actionName = getReadableActionName(action)
+
+    startCh.publish({ actionName, request })
+
+    callback = asyncResource.bind(callback)
```
Comment (bengl) [inline packages/datadog-instrumentations/src/sharedb.js:48]: Do you actually need this, given that you're wrapping everything on line 50 in a bind anyway?

Comment (bengl) [inline packages/datadog-instrumentations/src/sharedb.js:48]: Do you actually need this, given that you're wrapping everything in a bind on line 50 anyway?

File: packages/datadog-plugin-sharedb/src/index.js [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
+
+      if (this.config.hooks && this.config.hooks.receive) {
+        this.config.hooks.receive(span, request)
+      }
 
-function getReadableActionName (action) {
-  const actionName = READABLE_ACTION_NAMES[action]
-  if (actionName === undefined) {
-    return action
+      analyticsSampler.sample(span, this.config.measured)
```
Comment (bengl) [inline packages/datadog-plugin-sharedb/src/index.js:32]: Is this needed? The previous code doesn't seem to have used analytics sampler. Am I missing something here?

**Review by bengl — CHANGES_REQUESTED** [review]

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-instrumentations/src/sharedb.js (hunk 1 above)
Comment (bengl) [reply packages/datadog-instrumentations/src/sharedb.js:48]: Nevermind, this was explained to me.

**Review by rochdev — APPROVED** [review]

**Review by bengl — APPROVED** [review]

**Pull request merged**